src/libmach/elf.c

```diff
@@ -287,6 +289,9 @@ elfinit(int fd)
 	return e;
 
 err:
+	if (p != nil) {
```

Comment:
free(nil) is a no-op; why not just `free(p)` without the conditional?

Reply:
@dancrossnyc done as suggested